### PR TITLE
New DNS plugin: Alwaysdata

### PR DESCRIPTION
**Pull request by zopieux**
Thanks for providing us with certbot, that's a nice tool!

My DNS provider is [Alwaysdata](https://www.alwaysdata.com/) and it features an API to manage DNS records, which means it can be made compatible with certbot.

I took the time to write a DNS authenticator plugin for Alwaysdata, mimicking the other plugins as best as I could. I hope you'll find this plugin worthy to be added to certbot tree.

### Tests
All tests pass (tested py27 and py36). Pylint passes. ``certbot-dns-alwaysdata`` has 99% coverage.

### Structure
I split this PR in three commits:
* the main one, adding just ``certbot-dns-alwaysdata``
* a “certboot integration” commit, adding references to the module into certbot main tree but *not* the main documentation
* a “public integration” commit, adding references to the module into the documentation and the release tool. This one obviously cannot be merged as-is as it assumes that the PyPi, AUR and RTD packages exist. Feel free to cherry-pick it out, or modify it. For my own purposes, I [uploaded](https://pypi.org/project/certbot-dns-alwaysdata/) the package to PyPi, but I am 100% okay with sharing or transferring maintainership if this gets merged.

### CODE REVIEW

**Comment by Zopieux** [comment]
The build is failing because I'm using a `requests>=2.4.2` feature (`json=foo`). I don't understand how to fix this. I tried fiddling with `local-oldest-requirements.txt`, to no avail. Can you help?

File: certbot-dns-alwaysdata/certbot_dns_alwaysdata/__init__.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     -d www.example.com
+
+.. code-block:: bash
+   :caption: To acquire a certificate for ``example.com``, waiting 60 seconds
+             for DNS propagation
+
+   certbot certonly \\
+     --dns-alwaysdata \\
+     --dns-alwaysdata-credentials ~/.secrets/certbot/alwaysdata.ini \\
+     --dns-alwaysdata-propagation-seconds 60 \\
```
Comment (zjs) [inline certbot-dns-alwaysdata/certbot_dns_alwaysdata/__init__.py]: Could we also document this argument in the table above?

File: certbot-dns-alwaysdata/docs/api/dns_alwaysdata.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+:mod:`certbot_dns_alwaysdata.dns_alwaysdata`
+------------------------------------
```
Comment (zjs) [inline certbot-dns-alwaysdata/docs/api/dns_alwaysdata.rst]: Could you try generating the documentation (e.g., using `make html` from the `certbot-dns-alwaysdata/docs` directory with an active `venv`) and fixing any warnings/errors? I suspect you'll get one for this line of hyphens being too short. (ReStructured Text can be a bit picky about formatting.)

**Comment by Zopieux** [comment]
Thanks for the review. Do you also need a rebase against master? Is it too early?

**Comment by Zopieux** [comment]
mypy is not okay with me doing
```python
foo = (1, 2, 3)
bar = iter(foo)
# and then
next(bar)
```
and I have no idea why.

**Comment by Zopieux** [comment]
@bmw Gentle ping, can you please have another look? I've rebased against master and resolved the mypy warning. Waiting for your LGTM to update the setup.py version. Thanks.

**Comment by bmw** [comment]
Thanks @Zopieux. It's on my queue, we appreciate the PR, I'm sorry for the wait, and I hope to get to it as soon as I can, but to be honest it'll probably be a few more weeks as the small Certbot team is pretty swamped at the moment.

**Comment by yeupou** [comment]
Hi,

I was interested in using this plugin. But looks like it is not made for the current python version, is it?

**Comment by zopieux** [comment]
Feel free to take over this PR, I do not have the energy to work on updating this (otherwise very simple) code six years later.

**Pull request closed without merging**